torchvision/prototype/features/_bounding_box.py

```diff
@@ -128,13 +128,20 @@ def resized_crop(
         return BoundingBox.new_like(self, output, image_size=image_size, dtype=output.dtype)
 
     def pad(
-        self, padding: List[int], fill: Union[int, float, Sequence[float]] = 0, padding_mode: str = "constant"
+        self,
+        padding: Union[int, Sequence[int]],
+        fill: Union[int, float, Sequence[int], Sequence[float]] = 0,
```

Comment:
> Are there any BC/JIT considerations by changing the default value and type?

yes, I do not touch any stable public/private API with this except: https://github.com/pytorch/vision/pull/6245/files#diff-efb615f437f0b00ef2ed8abf2e0eda487cf609899a115bf3fa4691100db581afR307

Right now in the stable API, we have the following for fill default value:
```
-- Stable Transforms:

RandomAffine:
- fill=0

ElasticTransform:
- fill=0

RandomRotation:
- fill=0

RandomPerspective:
- fill=0

Pad:
- fill=0


-- Stable functional:

affine:
- fill: Optional[List[float]] = None,
-- pil: fill: Optional[Union[float, List[float], Tuple[float, ...]]] = 0,
-- t: fill: Optional[List[float]] = None

elastic_transform:
- fill: Optional[List[float]] = None,

rotate:
- fill: Optional[List[float]] = None,
-- pil: fill: Optional[Union[float, List[float], Tuple[float, ...]]] = 0,
-- t: fill: Optional[List[float]] = None,

perspective:
- fill: Optional[List[float]] = None,

pad:
- fill: Union[int, float] = 0
-- pil: fill: Optional[Union[float, List[float], Tuple[float, ...]]] = 0,
-- t: fill: Union[int, float] = 0,
```

We have two values: 0 and None. Supported fill types are numbers and sequences of numbers: affine, rotate started to support sequences about 1-2 years now and pad has a workaround introduced in prototype to support sequences.
Inspecting the code I do not see why keeping None is useful as fill is directly reassigned to 0.

> I understand that this layer is not JIT scriptable but the change on low-level kernels will expose differences to the users.

Yes, there is a change of default value from None -> 0 in low-level ops in proto which will become a replacement for current API. IMO, the change is minor and I think there is rare user who calls `pad(...., fill=None)`. If we agree that the change is necessary for code coherence etc I'm OK to add `Optional` and check for None and raise a warning about change. Otherwise, please also let me know if you want to revert default value change.

Reply:
From your response, I understand that the only challenge is for the users of the public `F` interface who currently pass None. If we are able to handle that in a BC way (by replacing None with 0 within the method), then I don't see any blocker. As long as we don't break any BC on public APIs, I'm in favour of aligning the calls. Do I miss anything?

Reply:
Restored default fill value to None